netmaster/master/api.go

```diff
@@ -88,6 +92,10 @@ func AllocAddressHandler(w http.ResponseWriter, r *http.Request, vars map[string
 
 	log.Infof("Received AddressAllocRequest: %+v", allocReq)
 
+	// Take a global lock for address allocation
+	addrMutex.Lock()
+	defer addrMutex.Unlock()
```

Comment:
just curious: for my understanding the locks are needed for various objects because multiple netmasters are allowed on the same cluster?